worker/download.go

```diff
@@ -0,0 +1,1088 @@
+package worker
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"sort"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+	rhpv2 "go.sia.tech/core/rhp/v2"
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/object"
+	"go.sia.tech/renterd/tracing"
+	"go.uber.org/zap"
+	"lukechampine.com/frand"
+)
+
+const (
+	maxConcurrentSectorsPerHost   = 3
+	maxConcurrentSlabsPerDownload = 3
```

Comment:
Is there some reasoning behind those numbers that you could leave here? I suppose you did some testing?

Reply:
Not really actually, both numbers just seemed like sane values. E.g. we want to buffer slabs, but not too much. We want to download sectors from a single host in parallel, I think having more than 3 concurrent sector downloads per host is a) unlikely to happen and b) unlikely it would be faster.

I don't mind doing some testing though